packages/frontend-2/components/project/page/acc/Syncs.vue

```diff
@@ -0,0 +1,536 @@
+    `https://developer.api.autodesk.com/data/v1/projects/${projectId}/versions/${encodeURIComponent(
+      versionId
+    )}/download`,
+    { headers: { Authorization: `Bearer ${tokens.value!.access_token}` } }
+  )
+  if (!res.ok) throw new Error(`Failed to generate ACC download URL`)
+  const { links } = await res.json()
+  return links.self.href
+}
+
+const onDownloadClick = async (item: AccItem) => {
+  try {
+    const signedUrl = await getSignedDownloadUrl(
+      selectedProjectId.value as string,
+      item.latestVersionId as string
+    )
+    window.open(signedUrl, '_blank')
+  } catch (e) {
+    triggerNotification({
+      type: ToastNotificationType.Danger,
+      title: 'Download failed',
+      description: e instanceof Error ? e.message : 'Unexpected error'
+    })
+  }
+}
+
+const onFileSelected = (item: AccItem) => {
+  selectedFolderContent.value = item
+}
+
+const { mutate: createAccSyncItem } = useMutation(accSyncItemCreateMutation)
+
+const addSync = async () => {
+  try {
+    // annoying but looks like ACC does not give the exact version number directly
+    const fileVersion = Number(
+      new URLSearchParams(
+        selectedFolderContent.value?.latestVersionId?.split('?')[1]
+      ).get('version')
+    )
+
+    const accFileViewName = revitViewName.value === '' ? undefined : revitViewName.value
+
+    await createAccSyncItem({
+      input: {
+        projectId: props.projectId,
+        modelId: selectedModel.value?.id as string,
+        accRegion: selectedHub.value?.attributes?.region as string,
+        accFileExtension: selectedFolderContent.value?.fileExtension as string,
+        accHubId: selectedHubId.value!,
+        accProjectId: selectedProjectId.value as string,
+        accRootProjectFolderUrn: rootProjectFolderId.value!,
+        accFileLineageUrn: selectedFolderContent.value?.id as string,
+        accFileName: (selectedFolderContent.value?.attributes.displayName ||
+          selectedFolderContent.value?.attributes.name) as string,
+        accFileVersionIndex: fileVersion,
+        accFileVersionUrn: selectedFolderContent.value?.latestVersionId as string,
+        accFileViewName
+      }
+    })
+    // TODO: NEED TO GO AWAY WHEN WE HAVE PROPER SUBSCRIPTIONS
+    setTimeout(() => {
+      refetchAccSyncItems()
+    }, 1000)
+  } catch (error) {
+    triggerNotification({
+      type: ToastNotificationType.Danger,
+      title: 'Add sync item failed',
+      description: error instanceof Error ? error.message : 'Unexpected error'
+    })
+  } finally {
+    selectedFolderContent.value = undefined
+    showNewSyncDialog.value = false
+    step.value = 0
+  }
+}
+
+const { mutate: deleteAccSyncItem } = useMutation(accSyncItemDeleteMutation)
+
+const handleDeleteSyncItem = async (id: string) => {
+  try {
+    await deleteAccSyncItem({
+      input: {
+        projectId: props.projectId,
+        id
+      }
+    })
+    // TODO: NEED TO GO AWAY WHEN WE HAVE PROPER SUBSCRIPTIONS
```

Comment:
dont we have subs? also, why do we need the timeout at all - the item isnt actually deleted after the mutation finishes?

Reply:
i will remove these timeouts, we have subs right now but havent connected yet to FE